src/shell/mod.rs

```diff
@@ -304,35 +304,10 @@ async fn execute_pipeline_inner(
   state: ShellState,
   stdin: ShellPipeReader,
   stdout: ShellPipeWriter,
-  mut stderr: ShellPipeWriter,
+  stderr: ShellPipeWriter,
 ) -> ExecuteResult {
   match pipeline {
     PipelineInner::Command(command) => {
-      // We only support redirects that are not in a pipe sequence
-      // at the moment, so this is fine to do only here
```

Comment:
When redirects and pipelines were initially implemented, the structure of what was parsed didn't exactly match shell semantics, which meant implementing this feature was hard. Now after serveral refactorings this was extremely trivial to fix/implement.